### PR TITLE
BFD-3179: Applied lombok annotations where possible under bfd-model

### PR DESCRIPTION
**Pull request by rcrupp**
**JIRA Ticket:**
[BFD-3179](https://jira.cms.gov/browse/BFD-3179)

**User Story or Bug Summary:**

As a BFD developer, I would like to consistently use lombok annotations in order to avoid writing redundant getter and setter methods for data classes. 

---

### What Does This PR Do?
* Adds Lombok `@Getter`, `@Setter`, `@NoArgsConstructor`, and `@AllArgsConstructor` annotations where appropriate.
* Updates classes in the `bfd-model-codebook-libary` and `bfd-model-rif` modules.

### What Should Reviewers Watch For?
If you're reviewing this PR, please check for these things in particular:
* Verify all PR security questions and checklists have been completed and addressed.
* When replacing actual getters/setters with the annotations the links to those methods in comments become broken so I had to remove the `{link}` around them. 
* I didn't add any `@ToString` annotations because the `ToString` functions that it would replace don't produce the equivalent output and I wasn't sure if there is tests/code somewhere that depends on the current text formatting.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist
I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-codebook-library/src/main/java/gov/cms/bfd/model/codebook/model/Value.java [hunk 1]
```diff
@@ -72,15 +51,6 @@ public String getDescription() {
...
-   *
-   * @param description the new value for {@link #getDescription()}
-   */
-  public void setDescription(String description) {
-    this.description = description;
-  }
-
   /** {@inheritDoc} */
   @Override
   public String toString() {
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-codebook-library/src/main/java/gov/cms/bfd/model/codebook/model/Value.java:56]: Not sure if we're using this toString to validate something and need it in this particular format, but lombok also has a toString annotation.

**Review by lsmitchell — DISMISSED** [review]
Nice reduction!

File: apps/bfd-model/bfd-model-codebook-library/src/main/java/gov/cms/bfd/model/codebook/model/Value.java (hunk 1 above)
Comment (rcrupp) [reply apps/bfd-model/bfd-model-codebook-library/src/main/java/gov/cms/bfd/model/codebook/model/Value.java]: Sounds good, I modified this file to use @ToString

**Review by lsmitchell — APPROVED** [review]

**Review by brandoncruz3 — APPROVED** [review]
👍 
image

**Pull request merged**